### PR TITLE
ci: switch to GitHub Actions

### PR DESCRIPTION
**Pull request by codebytere**
##### Description of change

This PR switches test builds from Travis to GitHub Actions.

cc @rvagg @cclauss 

##### Checklist

- [ ] `npm install && npm test` passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

**Comment by codebytere** [comment]
Not 100% sure why the windows tests are failing 🤔 i copied pretty directly from the travis file.

**Comment by cclauss** [comment]
Just use the same code that you use for the real operating systems.  That should work.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          python-version: ${{ matrix.python }}
+      - name: Set Env Vars
+        shell: bash
+        run: |
+          if [ "$PYTHON_VERSION" == 2.7 ]; then
+            if [ "$RUNNER_OS" == "Windows" ]; then
+              echo "::add-path::/c/Python27:/c/Python27/Scripts"
+              echo "::set-env name=NODE_GYP_FORCE_PYTHON::/c/Python27/python.exe"
+            else
+              echo "::set-env name=NODE_GYP_FORCE_PYTHON::python2"
```
Comment (cclauss) [inline .github/workflows/test.yml:39]:
```suggestion
            echo "::set-env name=NODE_GYP_FORCE_PYTHON::python2"
```

**Comment by cclauss** [comment]
On Windows you should be able to use `python` or maybe `py` or maybe `py3` to get to Python 3.

File: .github/workflows/test-python.yml [hunk 2]
```diff
@@ -29,12 +37,13 @@ jobs:
...
-        if: matrix.os == 'windows-latest'
+      - name: Test with pytest
         shell: bash
-        run: GYP_MSVS_VERSION=2015 GYP_MSVS_OVERRIDE_PATH="C:\\Dummy" pytest
+        run: |
+          if [ "$RUNNER_OS" == "Windows" ]; then
+            GYP_MSVS_VERSION=2015 GYP_MSVS_OVERRIDE_PATH="C:\\Dummy" python -m pytest
+          else
+            python -m pytest
+          fi
```
Comment (cclauss) [inline .github/workflows/test-python.yml:47]:
```suggestion
          python -m pytest
```
`actions/setup-python` will ensure that `python` points to `matrix.python-version` on all platforms.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      - name: Run Tests
+        shell: bash
+        run: |
+          echo "Testing with Python $(python --version 2>&1)"
+          npm test
+          if [ "$RUNNER_OS" == "Windows" ]; then
+            GYP_MSVS_VERSION=2015 GYP_MSVS_OVERRIDE_PATH="C:\\Dummy" python -m pytest
+          else
+            python -m pytest
+          fi
```
Comment (cclauss) [inline .github/workflows/test.yml:51]:
```suggestion
          python -m pytest
```
`actions/setup-python` will ensure that `python` points to `matrix.python-version` on all platforms.

File: .github/workflows/test-python.yml (hunk 2 above)
Comment (codebytere) [reply .github/workflows/test-python.yml:47]: ah, cool!

File: .github/workflows/test.yml (hunk 3 above)
Comment (codebytere) [reply .github/workflows/test.yml]: hmm - 
> `E         ValueError: Could not locate Visual Studio installation.`

this appears not to be the case.

**Comment by cclauss** [comment]
https://github.com/microsoft/setup-msbuild

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        with:
+          python-version: ${{ matrix.python }}
+        env:
+          PYTHON_VERSION: ${{ matrix.python }}
+      - name: Add msbuild to PATH
+        if: matrix.os == 'windows-latest'
+        uses: microsoft/setup-msbuild@v1.0.0
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip flake8 pytest==4.6.6
```
Comment (cclauss) [inline .github/workflows/test.yml:38]: Why hardcore pytest to a low version?

Comment (codebytere) [reply .github/workflows/test.yml:38]: i pulled it from the python examples used in stock python GitHub Actions. I'll remove it.

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    branches:
+      - master
+
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        node: [10.x, 12.x, 14.x]
+        python: [2.7, 3.6, 3.7, 3.8]
```
Comment (gengjiawen) [inline .github/workflows/test.yml:17]: Maybe remove 2.7 since it's EOL

**Pull request closed without merging**